### PR TITLE
Peer review changes - before authentication and carts rework

### PR DESCRIPTION
**Pull request by charm-13**
Fixes based on peer review changes listed in issues 8, 7, 9, and 10. Not including HTTP status codes and sanitizing inputs.

Formatted with "Black Formatter"

### CODE REVIEW

File: src/api/users.py [hunk 1]
```diff
@@ -85,3 +68,89 @@ def delete_user(user_id: int):
...
+    try:
+        with db.engine.begin() as connection:
+            adoptions = (
+                connection.execute(
+                    sqlalchemy.text(
+                        """
+                    SELECT 
+                        name, stage
+                    FROM user_creature_connection u
+                    LEFT JOIN creatures c ON u.creature_id = c.id
```
Comment (thuang-28) [inline src/api/users.py]: why a LEFT JOIN and not a JOIN? LEFT JOIN implies that it's possible for user_creature_connection to have creatures that don't exist in the creatures table

**Review by thuang-28 — CHANGES_REQUESTED** [review]
I was going to suggest a couple of nitpicky minor variable access things, but I feel like those should be handled after everything else, to be honest. as long as it all works, looks good!

**Review by samikshakarimbil — APPROVED** [review]

**Review by samikshakarimbil — CHANGES_REQUESTED** [review]
Make sure to put the new tables in schema.sql

File: src/api/users.py (hunk 1 above)
Comment (charm-13) [reply src/api/users.py]: True, I'll fix that 👍

**Review by thuang-28 — APPROVED** [review]
looks good

**Review by samikshakarimbil — APPROVED** [review]
Looks good!

**Pull request merged**